File: NetCore/ConsoleOop/ConsoleOop/Tasks/ITask.cs
using System.Collections.Generic;
using ConsoleOop.Configs;

namespace ConsoleOop.Tasks
{
    /// <summary>
    /// 執行備份類別
    /// </summary>
    interface ITask
    {
        /// <summary>
        /// 執行備份
        /// </summary>
        /// <param name="configs">檔案處理設定</param>
        /// <param name="schedules">自動排程時間</param>
        void Execute(List<Config> configs, List<Schedule> schedules);
    }
}

File: Laravel/app/Services/Tasks/ITask.php
<?php
/**
 * Created by PhpStorm.
 * User: tsao
 * Date: 2017/11/20
 * Time: 下午 08:48
 */

namespace App\Services\Tasks;


/**
 * 執行備份介面
 * Interface ITask
 * @author steve.tsao
 * @package App\Services\Tasks
 */
interface ITask
{
    /**
     * 執行備份
     * @author steve.tsao
     * @param array $configs 檔案處理設定
     * @param array $schedules 自動排程時間
     */
    public function Execute(array $configs, array $schedules);
}
